ipn/ipnlocal/local.go

```diff
@@ -935,8 +935,7 @@ func (b *LocalBackend) Start(opts ipn.Options) error {
 	httpTestClient := b.httpTestClient
 
 	if b.hostinfo != nil {
-		hostinfo.Services = b.hostinfo.Services // keep any previous session and netinfo
-		hostinfo.NetInfo = b.hostinfo.NetInfo
+		hostinfo.Services = b.hostinfo.Services // keep any previous services
 	}
 	b.hostinfo = hostinfo
```

Comment:
With the unlock/lock dance that happens higher up, it's very hard for me to assert the safety of this code.

Should the local hostinfo variable be instead intiialized only after the b.cc != nil check and subsequent lock recycle?

Reply:
At this point, the lock is still held. It is unlocked further down on line 979